Jenkinsfile

```diff
@@ -0,0 +1,273 @@
+                '<dt>-h</dt>\n' +
+                '<dd>this message</dd>\n' +
+                '<dt>-p gcc4</dt>\n' +
+                '<dd>primary compiler version [default: gcc7]</dd>\n' +
+                '<dt>-P password</dt>\n' +
+                '<dd>platform root password [default: randomly chosen]</dd>\n' +
+                '<dt>-r</dt>\n' +
+                '<dd>full strap build (no cache) [default: no]</dd>\n' +
+                '<dt>-S</dt>\n' +
+                '<dd>do *not* run smatch [default is to run smatch]</dd>\n' +
+                '<dt>-s gcc7</dt>\n' +
+                '<dd>shadow compilers, comma delimited (gcc4,gcc#) [default: none]</dd>\n' +
+                '</dl>'
+        )
+        text(
+            name: 'CONFIGURE_PROJECTS',
+            defaultValue:
+                'illumos-extra: master: origin\n' +
+                'illumos: master: origin\n' +
+                'local/kbmd: master: origin\n' +
+                'local/kvm-cmd: master: origin\n' +
+                'local/kvm: master: origin\n' +
+                'local/mdata-client: master: origin\n' +
+                'local/ur-agent: master: origin',
+            description:
+                'This parameter is used by smartos-live to decide ' +
+                'which branches to checkout and configure.</br>\n' +
+                'The string is formatted:<br/>' +
+                '<pre>\n' +
+                '&lt;relative path to ./projects&gt;:&lt;branch name&gt;:[optional git URL]\n' +
+                '</pre>' +
+                'In place of a full git url, the keyword \'origin\' ' +
+                'is allowed in order to specify the default github remote URL.'
+        )
+        // The default choice() is the first list item
+        choice(
+            name: 'PLATFORM_BUILD_FLAVOR',
+            choices: ['triton', 'smartos', 'triton-and-smartos'],
+            description:
+                '<p><dl>\n' +
+                '<dt>triton</dt>' +
+                  '<dd>the default, build a platform image and publish it</dd>\n' +
+                '<dt>smartos</dt>' +
+                  '<dd>build a platform image and smartos artifacts, but do ' +
+                  'not publish a Triton platform image</dd>\n' +
+                '<dt>triton-and-smartos</dt>' +
+                  '<dd>build both of the above</dd>\n' +
+                '</dl>' +
+                'The following are the SmartOS artifacts that will be ' +
+                'published when selecting one of the smartos options: ' +
+                '<ul>\n' +
+                '  <li>SmartOS iso image</li>\n' +
+                '  <li>SmartOS usb image</li>\n' +
+                '  <li>SmartOS vmware image</li>\n' +
+                '  <li>SmartOS Changelog file</li>\n' +
+                '</ul></p>'
+        )
+        booleanParam(
+            name: 'BUILD_STRAP',
+            defaultValue: false,
+            description: 'This parameter declares whether to build a new ' +
```

Comment:
This could probably do with a bit more clarification. We *always* build the strap, so this is perhaps not a good name for the variable. BUILD_STRAP_CACHE would be better,
and it should clarify that this is about building - and uploading - the strap cache used by devs, not the proto.strap that's used by Jenkins builds.

Reply:
That makes sense. I'll change that, updated the wording and modify illumos-extra accordingly.